### PR TITLE
Add OMIS pricing

### PR DESCRIPTION
**Pull request by marcofucci**
This:
- adds pricing fields to the order model
- updates those values when an order or an assignee is saved
- updates the order endpoints to return those values
- updates the search to include those fields. Only the total_cost is indexed as it will be used for searching in the near future
- updates the content of the quote to include the pricing values

### CODE REVIEW

File: datahub/omis/order/pricing.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    if order.vat_status == VATStatus.uk:
+        return True
+    if order.vat_status == VATStatus.outside_eu:
+        return False
+
+    assert order.vat_status == VATStatus.eu
+    return not order.vat_verified
+
+
+def _calculate_pricing(estimated_time, hourly_rate, vat_value, discount_value):
```
Comment (reupen) [inline datahub/omis/order/pricing.py]: It's hard to follow what the units are for these variables, could append to the names or add to the docstring?

File: datahub/omis/order/pricing.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    Calculate the pricing using the given base values.
+
+    :returns: pricing values
+    """
+    if estimated_time == 0:
+        return ZERO_PRICING
+
+    net_cost = round((estimated_time / 60) * hourly_rate)
+    subtotal_cost = max(net_cost - discount_value, 0)
+    vat_cost = round(float(vat_value) * subtotal_cost * 0.01)
```
Comment (reupen) [inline datahub/omis/order/pricing.py]: Seems like the rounding is doing two things:
1) Killing floating-point inaccuracies
2) Rounding fractional pennies

Is that right? For 2) is the rounding right in terms of VAT?

File: datahub/omis/order/test/test_pricing.py [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        """
+        Test that if the order doesn't have the right VAT fields populated,
+        it raises a ValidationError.
+        """
+        order = Order(vat_status=None)
+
+        with pytest.raises(ValidationError):
+            should_vat_be_applied(order)
+
+    def test_outside_eu_shouldnt(self):
```
Comment (reupen) [inline datahub/omis/order/test/test_pricing.py]: Could consider paramet(e)rise these.

File: datahub/omis/order/models.py [hunk 4]
```diff
@@ -149,10 +149,24 @@ class Order(BaseModel):
     )
     discount_value = models.PositiveIntegerField(default=0)
     discount_label = models.CharField(max_length=100, blank=True)
+
     vat_status = models.CharField(max_length=100, choices=VATStatus, blank=True)
     vat_number = models.CharField(max_length=100, blank=True)
     vat_verified = models.NullBooleanField()
 
+    net_cost = models.PositiveIntegerField(
+        default=0, help_text='Tot hours * hourly rate in pence.'
```
[the remark this answers is not in the document]
Comment (marcofucci) [reply datahub/omis/order/models.py]: Ha, true

File: datahub/omis/order/pricing.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+def can_pricing_be_calculated(order):
+    """
+    :returns: True if the pricing for `order` can be calculated
+        (e.g. all the fields required are filled in)
+    """
+    try:
+        _validate_vat(order)
+    except ValidationError:
+        return False
+    else:
```
[the remark this answers is not in the document]
Comment (marcofucci) [reply datahub/omis/order/pricing.py]: 👍

File: datahub/omis/order/pricing.py (hunk 1 above)
Comment (marcofucci) [reply datahub/omis/order/pricing.py]: 👍  for docstring

File: datahub/omis/order/pricing.py (hunk 2 above)
Comment (marcofucci) [reply datahub/omis/order/pricing.py]: That's correct.

For the VAT, `vat_value` is a `Decimal` so I need to convert it to float and then round up or down the final value.
I'm make sure our PO checks the values, I'm not too worried as the error could be max 1p and they can see the price before proceeding but better to double-check.

File: datahub/omis/order/test/test_pricing.py (hunk 3 above)
Comment (marcofucci) [reply datahub/omis/order/test/test_pricing.py]: yep good point
